src/conv/invokers/impl_gemm_dynamic.cpp

```diff
@@ -178,22 +178,20 @@ InvokerFactory MakeImplGemmDynamicBackwardDataInvokerFactory(const ProblemDescri
         const int gemm_k_gid = k * y_dot_slice_gid[gemm_id] * x_dot_slice_gid[gemm_id];
         is_gemm_not_empty.emplace_back(gemm_k_gid > 0);
     }
-    bool need_set_zero = true;
```

Comment:
I can wrap it in a workaround define like we do in other cases to keep the original if we are planning on coming back to this.

I also think I am going to swap to [this way of clearing](https://github.com/ROCm/MIOpen/blob/develop/src/include/miopen/solver/implicitgemm_ck_util.hpp#L744C1-L764C2) instead since I saw hipMemsetAsync giving better performance when it was an option.

Reply:
Sounds good!

I would've suggested ZeroOutTensor, but capturing the zero-set kernel time is desirable. If we have a canned solution for capturing async execution time, I don't know about it.

In this case it would be best to modify the kernel(s) to ensure every pixel is touched. Those sort of ops are often free due to being shadowed by global I/O latency.

Reply:
We have the [HipEventProfiler](https://github.com/ROCm/MIOpen/blob/9b4cf56919c85aae26a3eb9206b16dd9506c171b/src/include/miopen/hipoc_kernel.hpp#L50-L58) RAII wrapper that can do this.

I think ideally we would use this everywhere, and remove the automatic accumulation logic that happens in the handle, but that is a large change.

Reply:
Oh, interesting. I'm not sure it's real useful as-is--we would need explicit control of when `ResetKernelTime` is called and it needs to not block in the ctor, but it's a good starting point.

Reply:
HipEventProfiler will make a start event when constructed if profiling is enabled (nothing otherwise), and then a stop event when destructed if profiling is enabled, and then accumulate the difference between start / stop into the profiling result on the handle. Any ResetKernelTime calls in between should be removed, but also will not impact the overall result, as the start / stop events will capture the full profiling time within that scope, reset the profiling time, and then set the result to the difference between start / stop.

I think this isn't perfect, since the handle is also already capturing profile timing and synchronizing itself when it needs to, but HipEventProfiler ensures we capture all the kernel time used within a scope, and also prevents issues with the manual accumulation we do right now. I think It would be better if we only had the start / stop events being synchronized on, and used to capture the profiling results, but that is too big of a change for right now. 

Is there an issue with the above that I am missing? 
I think as is, HipEventProfiler is safe & correct for capturing the profiling results.